static/js/ducks/recentSources.js

```diff
@@ -0,0 +1,32 @@
+import messageHandler from "baselayer/MessageHandler";
+
+import * as API from "../API";
+import store from "../store";
+
+export const FETCH_RECENT_SOURCES = "skyportal/FETCH_RECENT_SOURCES";
+export const FETCH_RECENT_SOURCES_OK = "skyportal/FETCH_RECENT_SOURCES_OK";
+
+export const fetchRecentSources = () =>
+  API.GET("/api/internal/recent_sources", FETCH_RECENT_SOURCES);
+
+// Websocket message handler
+messageHandler.add((actionType, payload, dispatch) => {
+  if (actionType === FETCH_RECENT_SOURCES) {
+    dispatch(fetchRecentSources());
+  }
+});
+
+const reducer = (state = { recentSources: [] }, action) => {
+  switch (action.type) {
+    case FETCH_RECENT_SOURCES_OK: {
+      const recentSources = action.data;
```

Comment:
Could you specify what's in `action.data` explicitly?

Reply:
Do you mean to split `action.data` into more explicit properties? This is an array of objects instead of a single object so I'm not sure how to make this more explicit.

Reply:
Right, nm then